src/material-experimental/mdc-progress-spinner/progress-spinner.ts

```diff
@@ -0,0 +1,240 @@
+  mixinColor(MatProgressSpinnerBase, 'primary');
+
+/** Possible mode for a progress spinner. */
+export type ProgressSpinnerMode = 'determinate' | 'indeterminate';
+
+/**
+ * Base reference size of the spinner.
+ */
+const BASE_SIZE = 100;
+
+/**
+ * Base reference stroke width of the spinner.
+ */
+const BASE_STROKE_WIDTH = 10;
+
+@Component({
+  selector: 'mat-progress-spinner, mat-spinner',
+  exportAs: 'matProgressSpinner',
+  host: {
+    'role': 'progressbar',
+    'class': 'mat-mdc-progress-spinner mdc-circular-progress',
+    '[class._mat-animation-noopable]': `_noopAnimations`,
+    '[style.width.px]': 'diameter',
+    '[style.height.px]': 'diameter',
+    '[attr.aria-valuemin]': '0',
+    '[attr.aria-valuemax]': '100',
+    '[attr.aria-valuenow]': 'mode === "determinate" ? value : null',
+    '[attr.mode]': 'mode',
+  },
+  inputs: ['color'],
+  templateUrl: 'progress-spinner.html',
+  styleUrls: ['progress-spinner.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+})
+export class MatProgressSpinner extends _MatProgressSpinnerMixinBase implements AfterViewInit,
+  OnDestroy, CanColor {
+
+  /** Whether the _mat-animation-noopable class should be applied, disabling animations.  */
+  _noopAnimations: boolean;
+
+  /** Implements all of the logic of the MDC circular progress. */
+  _foundation: MDCCircularProgressFoundation;
+
+  /** The element of the determinate spinner. */
+  @ViewChild('determinateSpinner') _determinateCircle: ElementRef<HTMLElement>;
+
+  /** Adapter used by MDC to interact with the DOM. */
+    // TODO: switch to class when MDC removes object spread in foundation
+    // https://github.com/material-components/material-components-web/pull/6256
+  private _adapter: MDCCircularProgressAdapter = {
+    addClass: (className: string) => this._elementRef.nativeElement.classList.add(className),
+    hasClass: (className: string) => this._elementRef.nativeElement.classList.contains(className),
+    removeClass: (className: string) => this._elementRef.nativeElement.classList.remove(className),
+    removeAttribute: (name: string) => this._elementRef.nativeElement.removeAttribute(name),
+    setAttribute: (name: string, value: string) =>
+      this._elementRef.nativeElement.setAttribute(name, value),
+    getDeterminateCircleAttribute: (attributeName: string) =>
+      this._determinateCircle.nativeElement.getAttribute(attributeName),
+    setDeterminateCircleAttribute: (attributeName: string, value: string) =>
+      this._determinateCircle.nativeElement.setAttribute(attributeName, value),
+  };
+  constructor(public _elementRef: ElementRef<HTMLElement>,
+              @Optional() @Inject(ANIMATION_MODULE_TYPE) animationMode: string,
+              @Inject(MAT_PROGRESS_SPINNER_DEFAULT_OPTIONS)
+                defaults?: MatProgressSpinnerDefaultOptions) {
+    super(_elementRef);
+    this._noopAnimations = animationMode === 'NoopAnimations' &&
```

Comment:
If `defaults` is undefined we can't turn off animations? That seems wrong

Reply:
This is the same behaviour as the original MatProgressSpinner